radiant/mainframe.cpp

```diff
@@ -7808,6 +7824,20 @@ void MainFrame::OnFilterWorld(){
 	Sys_UpdateWindows( W_XY | W_CAMERA );
 }
 
+void MainFrame::OnFilterSky(){
+	GtkWidget *item = GTK_WIDGET( g_object_get_data( G_OBJECT( m_pWidget ), "menu_filter_sky" ) );
+	g_bIgnoreCommands++;
+	if ( ( g_qeglobals.d_savedinfo.exclude ^= EXCLUDE_SKY ) & EXCLUDE_SKY ) {
+		gtk_check_menu_item_set_active( GTK_CHECK_MENU_ITEM( item ), TRUE );
+	}
+	else{
+		gtk_check_menu_item_set_active( GTK_CHECK_MENU_ITEM( item ), FALSE );
+	}
+	g_bIgnoreCommands--;
+	PerformFiltering();
+	Sys_UpdateWindows( W_XY | W_CAMERA );
+}
+
```

Comment:
All of the changes to mainframe.cpp are to add the new sky filter to the menu, and to conditionalize some of the filter menus that do not pertain to idTech2, like patches. I'm not opposed to removing the conditionalizing if you find it offensive.